drivers/i2c/i2c_ll_stm32_v1.c

```diff
@@ -316,15 +318,29 @@ s32_t stm32_i2c_msg_write(struct device *dev, struct i2c_msg *msg,
 
 	if (msg->flags & I2C_MSG_RESTART) {
 		LL_I2C_GenerateStartCondition(i2c);
+		timeout = STM32_I2C_TIMEOUT_USEC;
 		while (!LL_I2C_IsActiveFlag_SB(i2c)) {
+			if (timeout <= 0) {
+				LL_I2C_GenerateStopCondition(i2c);
+				return -EIO;
+			}
+			k_busy_wait(1);
```

Comment:
What a bout increasing wait time.
Looping every usec might be excessive...

Reply:
Currently it is `1microsecond`.
How much should I do? 10, 100microsecond?
I think that if i increase the wait time, it can take a long time under normal circumstances.

Reply:
You're right and anyway, we'll loop less than initial implementation.
Just discard this comment.